internal/am/delete_transfer.go

```diff
@@ -0,0 +1,38 @@
+package am
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+
+	"github.com/artefactual-sdps/enduro/internal/sftp"
+)
+
+const DeleteTransferActivityName = "DeleteTransferActivity"
+
+type DeleteTransferActivityParams struct {
+	Destination string
+}
+
+type DeleteTransferActivity struct {
+	client sftp.Client
+	logger logr.Logger
+}
+
+func NewDeleteTransferActivity(logger logr.Logger, client sftp.Client) *DeleteTransferActivity {
+	return &DeleteTransferActivity{client: client, logger: logger}
+}
+
+func (a *DeleteTransferActivity) Execute(ctx context.Context, params *DeleteTransferActivityParams) error {
+	a.logger.V(1).Info("Execute DeleteTransferActivity",
+		"Destination", params.Destination,
+	)
+
+	err := a.client.Delete(ctx, params.Destination)
+	if err != nil {
+		a.logger.V(1).Error(err, fmt.Sprintf("Delete transfer activity failed for %s", params.Destination))
+	}
+
+	return nil
```

Comment:
I'd return a simple results struct here, just to confirm that the activity succeeded, e.g.

```
type DeleteTransferActivityResult struct {
	OK bool
}
```

Reply:
If the activity does not return an error than it succeeded. If we stick with it returning an error then this is unnecessary value.

Reply:
We should probably return an empty struct, e.g.: in https://github.com/temporalio/background-checks/blob/main/activities/activities.go they have four empty structs used as result types. I believe they do that to ease schema evolution, e.g. adding a new field to an empty struct is a backward-compatible change.